app/routes.py

```diff
@@ -0,0 +1,294 @@
+def handle_customer(cust_id):
+
+    id_error = Customer.validate_id(cust_id)
+
+    if id_error:
+        return id_error
+
+    customer = Customer.query.get(cust_id)
+    
+    if request.method == "GET":
+        
+        return customer.to_dict(), 200
+
+    elif request.method == "PUT":
+
+        request_body = request.get_json()
+
+        input_error = Customer.check_input_fields(request_body)
+
+        if input_error:
+            return input_error 
+
+        customer.name = request_body["name"]
+        customer.postal_code = request_body["postal_code"]
+        customer.phone = request_body["phone"]
+
+        db.session.commit()
+
+        return customer.to_dict(), 200
+
+    elif request.method == "DELETE":
+        
+        db.session.delete(customer)
+        db.session.commit()
+
+        return {
+            "id": customer.id,
+            "status": "deleted"
+        }
+
+@rentals_bp.route("/check-out", methods=["POST"])
+def handle_rental_checkout():
+
+    request_body = request.get_json()
+
+    # validate customer id
+    try: 
+        customer_id = request_body["customer_id"]
+        customer_id_error = Customer.validate_id(customer_id)
+        if customer_id_error:
+            return customer_id_error
+
+        # validate video id
+        video_id = request_body["video_id"]
+        video_id_error = Video.validate_id(video_id)
+        if video_id_error:
+            return video_id_error
+
+    except KeyError:
+        return { "message" : "missing information "}, 400
+
+    available_inventory = Rental.get_available_video_inventory(video_id)
+
+    if available_inventory == 0:
+        return { "message" : "Could not perform checkout"}, 400
+
+    # create new rental
+    new_rental = Rental.from_json(customer_id, video_id)
+
+    db.session.add(new_rental)
+    db.session.commit()
+
+    return new_rental.to_dict_check_out(), 200
+
+@customer_bp.route("/<cust_id>/rentals", methods=["GET"])
+def handle_customer_rentals(cust_id):
+    
+    id_error = Customer.validate_id(cust_id)
+
+    if id_error:
+        return id_error
+
+    if request.args:
+        params_error = validate_query_params(Rental, request)
+        if params_error:
+            return params_error
+        rentals = handle_query_params(Rental, request, {"customer_id": cust_id})
+    else:
+        rentals = Rental.query.filter_by(customer_id = cust_id)
+
+    return jsonify([rental.to_dict_customer_rentals() for rental in rentals]), 200
+
+
+@rentals_bp.route("/check-in", methods=["POST"])
+def rental_checkin_update():
+    request_body = request.get_json()
+    #checks responsebody
+    if "customer_id" not in request_body or "video_id" not in request_body:
+        return { "message" : "missing information "}, 400
+
+    #validate customer id
+    customer_id = request_body["customer_id"]
+    customer_id_error = Customer.validate_id(customer_id)
+    if customer_id_error:
+        return customer_id_error
```

Comment:
These validation steps are the same as checkout and are a great candidate for a helper function.